### ISSUE TITLE
!is operator?

### ISSUE THREAD

**Issue by nadako** [HaxeFoundation/haxe#9801]
Now when we have `is` together with `EIs`, maybe we should also have `!is` and `EIs(not:Bool)` or something.

The motivation for this is that negating `is` with normal `!` requires parens so it's rather ugly and annoying to write:
`if (!(something is SomeType))`
...compared to
`if (something !is SomeType)`

**Comment by back2dos** [comment]
I like it, but it'll have to be whitespace sensitive, because `something! is SomeType` is already valid. Perhaps the "should operators be white space sensitive?"-debate should be properly concluded before moving forward.

**Comment by nadako** [comment]
treating `!is` as kind of single token sounds fine to me. in reality we would probably just check for no space between two tokens, like we do in couple other places, but the outcome is the same.

that said, i'm wonderig if this ambiguity was the motivation for kotlin to have postfix `!!` rather than just `!`...

**Comment by Simn** [comment]
I find this pretty meh as far is syntactic sugar is concerned. I guess it's fine by itself, but I can already see someone asking about `!in` next and I'm not sure where all this takes us. Then again, the amount of "textual" operators is very limited in Haxe, so I suppose this is not much of an issue.

One thing to design is what `something!is Type` means. Is it `something! is Type` or `something !is Type`?

**Comment by RealyUniqueName** [comment]
Can we emit an error for such ambiguities?

**Comment by Simn** [comment]
I don't think so, this should be solved via precedence definitions. Otherwise you'd also have to error in cases like `a---b`.
